src/ngraph/ngraph_emitter.cc

```diff
@@ -1285,6 +1286,119 @@ void Emitter::CreateLayerOps() {
           data, ngraph::AxisSet{static_cast<size_t>(seq_axis)});
     }
   };
+  ngraph_op_funcs_["SoftmaxOutput"] = [this](const NodePtr& node) {
+    auto input = op_map_[node->inputs_[0]];
+    auto in_shape = input->get_shape();
+
+    ngraph::AxisSet axes;
+    if (get_default(node, "multi_output", false)) {
+      axes.insert(1);
+    } else if (get_default(node, "preserve_shape", false)) {
+      axes.insert(in_shape.size() - 1);
+    } else {
+      for (size_t i = 1; i < in_shape.size(); ++i) {
+        axes.insert(i);
+      }
+    }
+    return std::make_shared<ngraph::op::Softmax>(input, axes);
+  };
+}
+
+void Emitter::CreateLossOps() {
+  // These functions are in place to provide a mechanism for manually specifying
+  // backpropgation methods for Loss functions. We do this because MXNet
+  // provides a number of options that only effect the output of backprop, not
+  // forward prop, and are difficult or impossible to integrate into
+  // nGraph's autodiff functionality
+  loss_op_backward_funcs_["SoftmaxOutput"] = [this](
+      const NodePtr& node, const NgraphNodePtr& adjoint) {
+    float grad_scale = get_default(node, "grad_scale", 1.0f);
```

Comment:
Not relevant to this PR, but something we might discuss in a future navel-gazing exercise about coding style: Pros/cons of making these variables be `const`.

Reply:
In general, I want the inputs/outputs of a function to be const, but I don't think the temp variables need to be. Arguments the other way?